src/imgtools/utils/nnunet.py

```diff
@@ -1,103 +1,176 @@
-    :param dataset_reference: website of the dataset, if available
-    :param dataset_release:
-    :return:
+    # Define paths using pathlib
+    output_directory = pathlib.Path(output_directory)
+    shell_path = output_directory / 'train.sh'
+    base_dir = output_directory.parent.parent
+
+    if shell_path.exists():
+        shell_path.unlink()
+
+    # Define the environment variables and the script commands
+    script_content = f"""#!/bin/bash
+set -e
+
+export nnUNet_raw="{base_dir}/nnUNet_raw"
+export nnUNet_preprocessed="{base_dir}/nnUNet_preprocessed"
+export nnUNet_results="{base_dir}/nnUNet_trained_models"
+
+nnUNetv2_plan_and_preprocess -d {dataset_id} --verify_dataset_integrity -c 3d_fullres
+
+for (( i=0; i<5; i++ ))
+do
+    nnUNetv2_train {dataset_id} 3d_fullres $i
+done
+"""
+
+    # Write the script content to the file
+    with open(shell_path, "w", newline="\n") as f:
+        f.write(script_content)
+
+# Code take from: https://github.com/MIC-DKFZ/nnUNet/blob/master/nnunetv2/dataset_conversion/generate_dataset_json.py
+
+def generate_dataset_json(output_folder: str,
+                          channel_names: dict,
+                          labels: dict,
+                          num_training_cases,
+                          file_ending: str,
+                          regions_class_order: Tuple[int, ...] = None,
+                          dataset_name: str = None, reference: str = None, release: str = None, license: str = 'hands off!',
+                          description: str = None,
+                          overwrite_image_reader_writer: str = None, **kwargs):
     """
-    train_identifiers = get_identifiers_from_splitted_files(imagesTr_dir)
-
-    if imagesTs_dir is not None:
-        test_identifiers = get_identifiers_from_splitted_files(imagesTs_dir)
-    else:
-        test_identifiers = []
-
-    json_dict = {}
-    json_dict['name'] = dataset_name
-    json_dict['description'] = dataset_description
-    json_dict['tensorImageSize'] = "4D"
-    json_dict['reference'] = dataset_reference
-    json_dict['licence'] = license
-    json_dict['release'] = dataset_release
-    json_dict['modality'] = {str(i): modalities[i] for i in range(len(modalities))}
-    json_dict['labels'] = {str(i): labels[i] for i in labels.keys()}
-
-    json_dict['numTraining'] = len(train_identifiers)
-    json_dict['numTest'] = len(test_identifiers)
-    json_dict['training'] = [
-        {'image': "./imagesTr/%s.nii.gz" % i, "label": "./labelsTr/%s.nii.gz" % i} for i
-        in
-        train_identifiers]
-    json_dict['test'] = ["./imagesTs/%s.nii.gz" % i for i in test_identifiers]
-
-    if not output_file.endswith("dataset.json"):
-        print("WARNING: output file name is not dataset.json! This may be intentional or not. You decide. "
-              "Proceeding anyways...")
-    save_json(json_dict, os.path.join(output_file), sort_keys=sort_keys)
+    Generates a dataset.json file in the output folder
+
+    channel_names:
```

Comment:
num training cases missing type annotations

for the dict parameters, it would help to know a bit more of their type
something like Dict[str,str]


and for style help and easier readability, parameters could be on separate lines

also i think license could be a python built in so maybe be careful about that naming